### PR TITLE
solution to 15 - Hangman - 15

### LINKED ISSUE #36: 15 - Hangman-15
The following task is designed to introduce you to databases using sqlite3. In practice, what you will be implementing can be done in other ways and without databases but we'll use a database for learning purposes.

You'll be creating a table to store all guesses that a user makes.

Using sqlite modify your code so that:
- **A database table is created immediately the program starts up**
  - The name of the table should be 'Guesses'
  - The table should consist of one column named 'letter'
  - The column should only accept a single character
- **The table is dropped (deleted) when the game ends**

Note:
- The database code should be in a new module at `lib/database.py`. However you can call functions declared there from other modules in your code.
- the actual database file should be in the `data/` directory.

### PR DESCRIPTION
**Pull request by ebosetalee**
Resolves #36 [https://repl.it/join/nvgxaqit-emmanuellaesezo](https://repl.it/join/nvgxaqit-emmanuellaesezo)
[https://repl.it/@EmmanuellaEsezo/15-Hangman-15py-](https://repl.it/@EmmanuellaEsezo/15-Hangman-15py-)

Thank you for your patience.

### CODE REVIEW

File: hangman/lib/database.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+contents = sqlite3.connect("hangman/data/store_guesses.sqlite")
+contents.execute("CREATE TABLE IF NOT EXISTS guesses(letter CHAR)")
+
+def guesses_table(guess):
+    """ adds the guesses into the table created"""
+    contents.execute("INSERT INTO guesses VALUES('{}')".format(guess))
+    cursor = contents.cursor()
+    cursor.execute("SELECT * FROM guesses")
+    print(cursor.fetchall())
+    contents.commit()
```
Comment (yudori) [inline hangman/lib/database.py]: The question didn't ask for this function. You're to just  create and drop the table for now

File: hangman/lib/database.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import sqlite3
+
+
+contents = sqlite3.connect("hangman/data/store_guesses.sqlite")
```
Comment (yudori) [inline hangman/lib/database.py]: What's 'contents'? Use a better name

File: hangman/lib/database.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+contents.execute("CREATE TABLE IF NOT EXISTS guesses(letter CHAR)")
```
Comment (yudori) [inline hangman/lib/database.py]: put the logic to create the table in a separate function

File: hangman/lib/database.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    """ adds the guesses into the table created"""
+    contents.execute("INSERT INTO guesses VALUES('{}')".format(guess))
+    cursor = contents.cursor()
+    cursor.execute("SELECT * FROM guesses")
+    print(cursor.fetchall())
+    contents.commit()
+
+def drop_table():
+    """ deletes the table and its contents """
+    drops = "DROP TABLE guesses"
```
Comment (yudori) [inline hangman/lib/database.py]: use a better variable name

**Review by yudori — CHANGES_REQUESTED** [review]

File: hangman/lib/database.py (hunk 1 above)
Comment (ebosetalee) [reply hangman/lib/database.py]: So, I shouldn't insert values to the table?
@yudori

Comment (yudori) [reply hangman/lib/database.py]: Since you've already done it, just leave it.

File: hangman/lib/database.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+contents = sqlite3.connect("hangman/data/store_guesses.sqlite")
+contents.execute("CREATE TABLE IF NOT EXISTS guesses(letter CHAR)")
+
+def guesses_table(guess):
+    """ adds the guesses into the table created"""
+    contents.execute("INSERT INTO guesses VALUES('{}')".format(guess))
+    cursor = contents.cursor()
+    cursor.execute("SELECT * FROM guesses")
+    print(cursor.fetchall())
```
Comment (yudori) [inline hangman/lib/database.py]: You don't need these

File: hangman/lib/database.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import sqlite3
+
+
+contents = sqlite3.connect("hangman/data/store_guesses.sqlite")
+contents.execute("CREATE TABLE IF NOT EXISTS guesses(letter CHAR)")
+
+def guesses_table(guess):
```
Comment (yudori) [inline hangman/lib/database.py]: This function stores a guessed letter. Improve the naming here

**Review by yudori — CHANGES_REQUESTED** [review]

File: hangman/lib/database.py (hunk 6 above)
Comment (ebosetalee) [reply hangman/lib/database.py]: if i use guessed_letter, I'll have issues with naming when calling the file in core.py

File: hangman/lib/database.py (hunk 5 above)
Comment (ebosetalee) [reply hangman/lib/database.py]: ok

File: hangman/lib/database.py (hunk 6 above)
Comment (yudori) [reply hangman/lib/database.py]: It's an action, you can include that in the name. `store_guessed_letter`

Comment (ebosetalee) [reply hangman/lib/database.py]: ok

**Comment by ebosetalee** [comment]
I'm thinking of putting drop_table() in main.py

**Comment by ebosetalee** [comment]
I'm thinking of putting drop_table() in main.py 
@yudori

**Comment by ebosetalee** [comment]
@yudori

File: hangman/lib/database.py [hunk 7]
```diff
@@ -7,9 +7,9 @@ def create_table():
     """sqlite table created to store guesses"""
     db.execute("CREATE TABLE IF NOT EXISTS guesses(letter CHAR)")
 
-def guesses_table(stored_guessed_letter):
+def guesses_table(store_guessed_letter):
```
Comment (yudori) [inline hangman/lib/database.py]: No. I mean rename the function. The variable  name was ok as `guess`.

Comment (ebosetalee) [reply hangman/lib/database.py]: @yudori done

**Review by yudori — APPROVED** [review]

**Review by jirevwe — APPROVED** [review]

**Pull request merged**